spec/models/metric_spec.rb

```diff
@@ -1107,6 +1107,7 @@
         @vms_in_az = []
         2.times { @vms_in_az << FactoryGirl.create(:vm_openstack, :ems_id => @ems_openstack.id) }
         @availability_zone.vms = @vms_in_az
+        @availability_zone.vms.push(FactoryGirl.create(:vm_openstack, :ems_id => nil))
```

Comment:
Ah, I think this must be because of [`VmOrTemplate#disconnect_inv`](https://github.com/ManageIQ/manageiq/blob/6f8773c8205d0d36d5ca914a51c92e439987f59d/app/models/vm_or_template.rb#L666).  That method still looks very "infra" specific.  I guess we need to override that method in `CloudManager::Vm` and `InfraManager::Vm` in order to do things like `disconnect_host` and `disconnect_availability_zone` only where appropriate.

Nice find @kbrock